department-of-veterans-affairs/caseflow
# Resolve POA information when Substitute Appellant does not have a POA

Resolves [Jira ticket 1753](https://vajira.max.gov/browse/CASEFLOW-1753).

### Description
If a dependent has no POA, that dependent should still be able to be selected as a substitute appellant.  This PR makes that possible by removing power of attorney as a requirement for creating a substitute appellant.  Additionally, when a substitute appellant has a power of attorney, that information remains visible in case details.

### Acceptance Criteria
- [ ] In the substitute appellant flow, user can select as claimant a dependent who has no power of attorney 

### Testing Plan
1. Run the new migration in `db/migrate/20210622145703_remove_non_null_constraint_from_poa_id.rb`.
2. Generate an appeal for testing.
````
RequestStore[:current_user] = User.system_user
appeal = FactoryBot.create(:appeal, :dispatched_with_decision_issue)
````
3. Proceed through the appellant substitution flow and select the child “Tom Brady” as appellant.  This claimant does not have a POA in our fake data. 
<img width="1792" alt="SelectTomBrady" src="https://user-images.githubusercontent.com/80347702/123117602-e36b7380-d40f-11eb-9f72-d615a7de9aaa.png">



4. Ensure that there is no power of attorney on the case details page.  Should see the below for “Appellant’s Power of Attorney.”
<img width="1792" alt="Screen Shot 2021-06-22 at 12 30 00 PM" src="https://user-images.githubusercontent.com/80347702/123117849-19105c80-d410-11eb-9f90-2869ebfd3ea2.png">

5. Ensure that you don’t get any errors when you click the “refresh POA” button.


6. Roll back the migration and ensure that nothing breaks.  The `down` component of this migration should change any null values for `poa_participant` id to be empty strings.

### User Facing Changes
 - [ ] If there is no power of attorney, a "Successfully refreshed. No power of attorney information was found at this time." message will display when the user clicks the "refresh POA" button.  Previously in this scenario, the button would hang indefinitely. 
<img width="898" alt="Screen Shot 2021-06-25 at 1 24 55 PM" src="https://user-images.githubusercontent.com/80347702/123462970-f3728700-d5b8-11eb-99ab-1dc0bc054b4d.png">


### Database Changes
*Only for Schema Changes*

* ~[ ] Add typical timestamps (`created_at`, `updated_at`) for new tables~ No new tables.
* [X] Update column comments
* [X] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
~* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)~ n/a, no columns were changed
~* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)~
* [X] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb`
* [X] Run `make docs` (after running `make migrate`) to update DB schema docs
~* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)~
